### PR TITLE
Fix transaction ID implementation and verify Event.body is indeed base64 when isBase64 is true. 

### PR DESCRIPTION
**Pull request by selectiveduplicate**
Will bump the version when we approve of the chanegs. :slightly_smiling_face: 
## Screenshots from a working example with Serverless and the current patch

<details>
  <summary>See screenshot</summary>
  
</details>

<details>
  <summary>See screenshot</summary>
  
</details>

## Output from the added local tests

```sh
;; node --test
▶ testing response for async/await Lambda
  ✔ should contain 'X-Moesif-Transaction-Id' header (0.960301ms)
  ✔ should have 'application/json' as the 'Content-Type' header (0.501034ms)
  ✔ should have expected body (0.20235ms)
  ✔ should have '200 OK' status code (0.203222ms)
▶ testing response for async/await Lambda (3.188207ms)
{
  userId: 1,
  id: 1,
  title: 'sunt aut facere repellat provident occaecati excepturi optio reprehenderit',
  body: 'quia et suscipit\n' +
    'suscipit recusandae consequuntur expedita et cum\n' +
    'reprehenderit molestiae ut ut quas totam\n' +
    'nostrum rerum est autem sunt rem eveniet architecto'
}
▶ testing response for callback-style Lambda
  ✔ should contain 'X-Moesif-Transaction-Id' header (0.957489ms)
  ✔ should have 'application/json' as the 'Content-Type' header (0.483045ms)
  ✔ should have expected body (0.191357ms)
  ✔ should have '200 OK' status code (0.171779ms)
▶ testing response for callback-style Lambda (3.15757ms)
{
  userId: 1,
  id: 1,
  title: 'sunt aut facere repellat provident occaecati excepturi optio reprehenderit',
  body: 'quia et suscipit\n' +
    'suscipit recusandae consequuntur expedita et cum\n' +
    'reprehenderit molestiae ut ut quas totam\n' +
    'nostrum rerum est autem sunt rem eveniet architecto'
}
ℹ tests 8
ℹ suites 2
ℹ pass 8
ℹ fail 0
ℹ cancelled 0
ℹ skipped 0
ℹ todo 0
ℹ duration_ms 1057.238274
```

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -157,16 +158,29 @@ module.exports = function (options, handler) {
...
+    let extraHeaders = {
+      'X-Moesif-Transaction-Id': responseTransactionId,
+    };
+
     logMessage(options.debug, 'moesifMiddleware', 'start');
+
     var next = function (err, result) {
-      logEvent(event, context, err, result, options, moesifController);
-      callback(err, result)
+      const modifiedResultObject = {
```
Comment (xinghengwang) [inline lib/index.js:171]: please add more safety checks, what is result is null?

File: lib/index.js [hunk 2]
```diff
@@ -157,16 +158,29 @@ module.exports = function (options, handler) {
   var moesifMiddleware = function (event, context, callback) {
     moesifConfigManager.tryGetConfig();
 
+    const responseTransactionId = event.headers.transactionID || uuid4();
+
```
Comment (xinghengwang) [inline lib/index.js:163]: please add bit more safety. what if event.headers is null?

**Review by xinghengwang — CHANGES_REQUESTED** [review]

File: lib/index.js (hunk 2 above)
Comment (selectiveduplicate) [reply lib/index.js:163]: @xinghengwang Done.
Just wondering, we check for existing txn ID with `event.headers.transactionID` like above. Should we do it like `event.headers['X-Moesif-Transaction-Id']` instead?

Comment (xinghengwang) [reply lib/index.js:163]: yes. we should use event.headers['X-Moesif-Transaction-Id']

**Review by xinghengwang — APPROVED** [review]

**Pull request merged**